src/tools/android/java/com/google/devtools/build/android/r8/Desugar.java

```diff
@@ -635,10 +635,40 @@ private static void validateOptions(DesugarOptions options) {
         options.outputJars.size());
   }
 
-  public static void main(String[] args) throws Exception {
-    DesugarOptions options = parseCommandLineOptions(args);
-    validateOptions(options);
+  private static void runPersistentWorker() throws IOException {
+    while (true) {
```

Comment:
I mostly cribbed this from the legacy desugarer, which I think may have been written before using WorkRequestHandler was the norm. I'll take a look into this.

Reply:
I have an internal change that I can put a PR up for that converts the legacy desugarer over to `WorkRequestHandler` with multiplex support. But may not be worth it since I assume that the legacy implementation will be going away.

Reply:
Yes, we do intend to obviate the legacy desugarer once the D8 version is fully implemented and vetted. Exciting stuff!

I've updated the coding style in r8/Desugar.java to use WorkRequestHandler as you've suggested. Thanks as always!

I do think that I'll eventually need to add more specialized try/catch blocks for specific error scenarios; I'm sure some of those scenarios will come to light once we start pushing this code harder.